src/MongoDB.Driver.Core/Core/Operations/RetryableReadContext.cs

```diff
@@ -0,0 +1,179 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Misc;
+
+namespace MongoDB.Driver.Core.Operations
+{
+    /// <summary>
+    /// Represents a context for retryable reads.
+    /// </summary>
+    /// <seealso cref="System.IDisposable" />
+    public sealed class RetryableReadContext : IDisposable
```

Comment:
This class is almost identical to RetryableWriteContext just replacing "write" with "read".